src/components/NewClusterModal.js

```diff
@@ -149,6 +150,66 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
     ]))
   }
 
+  updateCluster(isStopRequired = false) {
+    const { currentCluster, onSuccess } = this.props
+    const { googleProject, clusterName } = currentCluster
+
+    if (isStopRequired) {
+      notify('success', 'To be updated, your runtime will now stop, and then start.', { timeout: 3000 })
+    }
+
+    return onSuccess(
+      Ajax().Clusters.cluster(googleProject, clusterName).update({
+        machineConfig: this.getMachineConfig()
+      }),
+      5000)
```

Comment:
here, I wait 5 seconds before refreshing if we are going to stop. We cannot refresh immediately, because leonardo needs to do a bit of work before the cluster status is actually updating the cluster status in this specific case. In my testing I found 5 seconds a good balance. The only consequence to missing the window, is, if the user isn't actually editting their notebook yet, the cluster can transition from 'Running' to 'Starting', and this case is covered by the notification on line 158.